### ISSUE TITLE
qt6 support

### ISSUE THREAD

**Issue by nyx0** [Jinmo/ifred#17]
awesome tool!
do you plan to add support for qt6?

**Comment by Jinmo** [comment]
My primary target to support was IDA Pro which uses Qt 5.6, and I believe there might be better command palettes for general purpose usage. So it's unlikely that I'll push code in near future..

Do you have any projects in mind that uses Qt 6 and needs a command palette?

**Comment by nyx0** [comment]
no project in particular, I upgraded to IDA 7.6 and macOS 11 and it's not working... Since I became addict to ifred. Not having it is a big downside in my workflow, so I tried to compile it (brew install qt 6 by default). I think I managed to update some code that is deprecated in qt6 but overall it's not compiling...
Maybe a better plan is to ask you if it's possible to have binary for IDA7.6 on macOS 11 :)

**Comment by Jinmo** [comment]
Sorry, I don't have binary for IDA 7.6 on macOS 11.. I thought it was still using Qt 5.6.3. Would you mind sharing the error log occurred when compiling it?

**Comment by nyx0** [comment]
sure, this is my error:
```
```

**Comment by Jinmo** [comment]
Thank you! Can you specify IDA_INSTALL_DIR as /Applications/IDAPro7.6/Contents/MacOS? Also I've uploaded some fix for warnings.

**Comment by nyx0** [comment]
Hello,
sorry for the late response.
I tried your updated standalone but I got this error that was already mention in previous issue:
```
dlopen(/Users/user/.idapro/plugins/ida_palette.dylib): dlopen(/Users/user/.idapro/plugins/ida_palette.dylib, 2): Library not loaded: @executable_path/plugins/python.dylib
  Referenced from: /Users/user/.idapro/plugins/ida_palette.dylib
  Reason: image not found
/Users/user/.idapro/plugins/ida_palette.dylib: can't load file
```
I know that it *should* works but it's not :(

Just to be sure here are the standalone I downloaded:
```bash
$ xattr -d com.apple.quarantine .idapro/plugins/ida_palette*
$ shasum .idapro/plugins/ida_palette*
431f33018168cfdee77e11d1fcade59aaec00c17  .idapro/plugins/ida_palette.dylib
2f8fe3e9a56bbc98f970615d6f53b84192fb3917  .idapro/plugins/ida_palette64.dylib
```

For compiling I created a small PR #18 because I don't have those filenames. I checked and it looks like on Windows it's the same (filenames).
I tried to compile but got this:
```bash
$ export IDA_INSTALL_DIR=/Applications/IDAPro7.6/ida.app/Contents/MacOS
$ export IDA_IDA_LIBRARY=/Applications/IDAPro7.6/ida.app/Contents/MacOS
$ mkdir build
$ cd build
$ cmake ../ \
    -DCMAKE_PREFIX_PATH=$(brew --prefix qt5) \
    -DIDA_SDK=$IDA_SDK \
    -DIDA_BINARY_64=True \
    -DIDA_EA_64=False \
    -GNinja \
    -DPYBIND11_PYTHON_VERSION=3 \
    -DIDA_INSTALL_DIR=$IDA_INSTALL_DIR \
    -DCMAKE_BUILD_TYPE=RelWithDebInfo
ninja

-- The C compiler identification is AppleClang 12.0.5.12050022
-- The CXX compiler identification is AppleClang 12.0.5.12050022
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working C compiler: /Library/Developer/CommandLineTools/usr/bin/cc - skipped
-- Detecting C compile features
-- Detecting C compile features - done
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Check for working CXX compiler: /Library/Developer/CommandLineTools/usr/bin/c++ - skipped
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- pybind11 v2.6.3 dev1
-- Found PythonInterp: /usr/local/bin/python3 (found suitable version "3.9.5", minimum required is "3") 
-- Found PythonLibs: /usr/local/opt/python@3.9/Frameworks/Python.framework/Versions/3.9/lib/libpython3.9.dylib
-- Performing Test HAS_FLTO
-- Performing Test HAS_FLTO - Success
-- Performing Test HAS_FLTO_THIN
-- Performing Test HAS_FLTO_THIN - Success
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_VISIBILITY - Success
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY
-- Performing Test COMPILER_HAS_HIDDEN_INLINE_VISIBILITY - Success
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR
-- Performing Test COMPILER_HAS_DEPRECATED_ATTR - Success
-- Configuring done
-- Generating done
-- Build files have been written to: /Users/user/Documents/ifred/build
[14/23] Linking CXX static library palette/libpalette.a
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: palette/libpalette.a(action.cpp.o) has no symbols
/Library/Developer/CommandLineTools/usr/bin/ranlib: file: palette/libpalette.a(action.cpp.o) has no symbols
[21/23] Linking CXX executable standalone/standalone
FAILED: standalone/standalone 
: && /Library/Developer/CommandLineTools/usr/bin/c++ -O2 -g -DNDEBUG -isysroot /Library/Developer/CommandLineTools/SDKs/MacOSX11.3.sdk -Wl,-search_paths_first -Wl,-headerpad_max_install_names  standalone/CMakeFiles/standalone.dir/standalone_autogen/mocs_compilation.cpp.o standalone/CMakeFiles/standalone.dir/standalone.cpp.o -o standalone/standalone  -Wl,-rpath,/Applications/IDAPro7.6/ida.app/Contents/MacOS/../Frameworks  /usr/local/opt/qt@5/lib/QtWidgets.framework/QtWidgets  palette/libpalette.a  /usr/local/opt/qt@5/lib/QtGui.framework/QtGui  /usr/local/opt/qt@5/lib/QtCore.framework/QtCore  /Applications/IDAPro7.6/ida.app/Contents/MacOS/../Frameworks/QtWidgets.framework/QtWidgets  /Applications/IDAPro7.6/ida.app/Contents/MacOS/../Frameworks/QtGui.framework/QtGui  /Applications/IDAPro7.6/ida.app/Contents/MacOS/../Frameworks/QtCore.framework/QtCore && :
Undefined symbols for architecture x86_64:
  "QT::QAbstractItemView::eventFilter(QT::QObject*, QT::QEvent*)", referenced from:
      vtable for PaletteItems in libpalette.a(items.cpp.o)
ld: symbol(s) not found for architecture x86_64
clang: error: linker command failed with exit code 1 (use -v to see invocation)
[22/23] Building CXX object palette/src/ida/CMakeFiles/ida_palette.dir/plugin.cpp.o
ninja: build stopped: subcommand failed.
```

thanks again for your help and time

**Comment by Jinmo** [comment]
Ah I see.. It seems like the filename of IDAPython plugin was changed in 7.4. Thank you for the PR! Now [the binaries](https://jinmo123.visualstudio.com/idapkg/_build/results?buildId=190&view=artifacts&pathAsName=false&type=publishedArtifacts) from azure pipeline should work as well.

Since the standalone builds were only used as an example, I disabled it for now.
It was also not compatible with normal Qt libraries, so I guess building it separately might be easier for everyone...
